packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -248,10 +304,79 @@ - (void)errorNoPhotoAccess:(PHAuthorizationStatus)status {
 
 - (void)showPhotoLibrary {
   // No need to check if SourceType is available. It always is.
-  _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
-  [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
-                                                    animated:YES
-                                                  completion:nil];
+  if (__phPickerFlag) {
+    [[self viewControllerWithWindow:nil] presentViewController:__pickerViewController
+                                                      animated:YES
+                                                    completion:nil];
+  } else {
+    _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
+                                                      animated:YES
+                                                    completion:nil];
+  }
+}
+
+// Limited access to the photo library
+- (void)showLimitedPhotoLibrary {
+  [[PHPhotoLibrary sharedPhotoLibrary]
+      presentLimitedLibraryPickerFromViewController:__pickerViewController];
+}
+
+- (void)picker:(PHPickerViewController *)picker
+    didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+  [picker dismissViewControllerAnimated:YES completion:nil];
+
+  for (PHPickerResult *result in results) {
+    [result.itemProvider
+        loadObjectOfClass:[UIImage class]
+        completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
+                            NSError *_Nullable error) {
+          if ([image isKindOfClass:[UIImage class]]) {
+            if (image != nil) {
+              NSNumber *maxWidth = [self->__arguments objectForKey:@"maxWidth"];
+              NSNumber *maxHeight = [self->__arguments objectForKey:@"maxHeight"];
+              NSNumber *imageQuality = [self->__arguments objectForKey:@"imageQuality"];
+
+              if (![imageQuality isKindOfClass:[NSNumber class]]) {
+                imageQuality = @1;
+              } else if (imageQuality.intValue < 0 || imageQuality.intValue > 100) {
+                imageQuality = [NSNumber numberWithInt:1];
+              } else {
+                imageQuality = @([imageQuality floatValue] / 100);
+              }
+
+              if (maxWidth != (id)[NSNull null] || maxHeight != (id)[NSNull null]) {
+                image = [FLTImagePickerImageUtil scaledImage:image
+                                                    maxWidth:maxWidth
+                                                   maxHeight:maxHeight];
+              }
+
```

Comment:
So we can refactor this part into 2 methods like: 
```objc
-(UIImage *)scaleImageIfNecessary:(UIImage *)image {
  NSNumber *maxWidth = [weakSelf.arguments objectForKey:@"maxWidth"];
  NSNumber *maxHeight = [weakSelf.arguments objectForKey:@"maxHeight"];
  if (maxWidth != (id)[NSNull null] || maxHeight != (id)[NSNull null]) {
  image = [FLTImagePickerImageUtil scaledImage:image
                                                    maxWidth:maxWidth
                                                   maxHeight:maxHeight];
  }
  return image;
}
```

and 

```objc
- (NSNumber *)getDesiredImageQuality {
   NSNumber *imageQuality = [weakSelf.arguments objectForKey:@"imageQuality"];
   if (![imageQuality isKindOfClass:[NSNumber class]]) {
                imageQuality = @1;
   } else if (imageQuality.intValue < 0 || imageQuality.intValue > 100) {
                imageQuality = [NSNumber numberWithInt:1];
   } else {
                imageQuality = @([imageQuality floatValue] / 100);
  }
  return imageQuality;
}
```

Reply:
I implemented `getDesiredImageQuality `. But I am not sure about `scaleImageIfNecessary ` since we use `maxWidth ` and  
`maxHeight ` also in the `else` block above the code.